tools/breaking-change-detector/rules/rules_field.go

```diff
@@ -14,29 +14,173 @@ type FieldRule struct {
+	fieldRule_ChangingType,
 	fieldRule_BecomingRequired,
+	fieldRule_BecomingComputedOnly,
+	fieldRule_OptionalComputedToOptional,
+	fieldRule_DefaultModification,
+	fieldRule_GrowingMin,
+	fieldRule_ShrinkingMax,
+	fieldRule_ChangingFieldDataFormat,
+}
+
+var fieldRule_ChangingFieldDataFormat = FieldRule{
+	name:       "Changing field data format",
+	definition: "Modification of the data format (either by the api or manually) will cause a diff with any sequential plans if that field is not Computed. Thus resulting in a breakage. Api breaking changes are out of scope with respect to provider responsibility but we may make changes in response to breakages in some instances to provide more customer stability.",
+	identifier: "field-changing-type",
+}
+
+var fieldRule_ChangingType = FieldRule{
+	name:        "Changing Field Type",
+	definition:  "While Certain Field Type migrations may be supported at a technical level it's a practice that we highly discourage. We see little value for these transitions vs the risk they impose.",
+	message:     "Field {{field}} changed from {{oldType}} to {{newType}} on {{resource}}",
+	identifier:  "field-changing-type",
+	isRuleBreak: fieldRule_ChangingType_func,
+}
+
+func fieldRule_ChangingType_func(old, new *schema.Schema, mc MessageContext) string {
+	message := mc.message
+	if old.Type != new.Type {
+		oldType := getValueType(old.Type)
+		newType := getValueType(new.Type)
+		message = strings.ReplaceAll(message, "{{oldType}}", oldType)
+		message = strings.ReplaceAll(message, "{{newType}}", newType)
+		return populateMessageContext(message, mc)
+	}
+
+	oldCasted, _ := old.Elem.(*schema.Schema)
+	newCasted, _ := new.Elem.(*schema.Schema)
+	if oldCasted != nil && newCasted != nil && oldCasted.Type != newCasted.Type {
+		oldType := getValueType(old.Type) + "." + getValueType(oldCasted.Type)
+		newType := getValueType(new.Type) + "." + getValueType(newCasted.Type)
+		message = strings.ReplaceAll(message, "{{oldType}}", oldType)
+		message = strings.ReplaceAll(message, "{{newType}}", newType)
+		return populateMessageContext(message, mc)
+	}
+
+	return ""
 }
 
 var fieldRule_BecomingRequired = FieldRule{
 	name:        "Field becoming Required Field",
-	definition:  "A field cannot become required as existing terraform modules may not have this field defined. Thus breaking their modules in sequential plan or applies.",
+	definition:  "A field cannot become required as existing terraform modules may not have this field defined. Thus breaking their modules in sequential plan or applies. If you are adding required to a field so a block won't remain empty this can cause two issues. First if it's a singular nested field the block may gain more fields later and it's not clear whether the field is actually required so it may be misinterpreted by future contributors. Second if users are defining empty blocks in existing configurations this change will break them. Consider these points in admittance of this type of change.",
```

Comment:
```suggestion
	definition:  "A field cannot become required as existing configs may not have this field defined. Thus, breaking configs in sequential plan or applies. If you are adding Required to a field so a block won't remain empty, this can cause two issues. First if it's a singular nested field the block may gain more fields later and it's not clear whether the field is actually required so it may be misinterpreted by future contributors. Second if users are defining empty blocks in existing configurations this change will break them. Consider these points in admittance of this type of change.",
```